gap/io.gi

```diff
@@ -1390,22 +1457,605 @@ function(str)
+                    Info(InfoWarning, 1, StringFormatted(
+                      Concatenation("Ignoring illegal edge ({}, {}) ",
+                      "(vertices {}-indexed, on line {}) "),
+                      v + r.labelorg - 1,
+                      w + r.labelorg - 1,
+                      r.labelorg,
+                      r.newline));
+                    else
+                        AddSet(r.edgeList[v], w);
+                        if not r.digraph then
+                            AddSet(r.edgeList[w], v);
+                        fi;
+                    fi;
+                fi;
+            fi;
+        elif c = ';' then
+              neg := false;
+              v := v + 1;
+            if v > r.n then
+              return;
+            fi;
+        elif c in "?\n" then
+            neg := false;
+        elif c = '.' then
+            return;
+        elif c = '-' then
+            neg := true;
+        elif c = '!' then
+            while c <> '\n' and c <> fail do
+                c := DIGRAPHS_GetUngetChar(r, D, 1);
+            od;
+        else
+          if c = fail then
+            return;
+          else
+            ErrorNoReturn("Illegal character ", c, " on line ", r.newline);
+          fi;
+        fi;
+    od;
+end);
+
+# helper function for DigraphFromDreadnautString
+# returns the next full integer from the string,
+# allowing for an optional '=' before the integer
+BindGlobal("DIGRAPHS_GetInt",
+function(r, D)
+    local ch;
+
+    ch := DIGRAPHS_GETNWCL(r, D, " \n\t\r");
+    if ch <> '=' then
+        DIGRAPHS_GetUngetChar(r, D, -1);
+    fi;
+    return DIGRAPHS_readinteger(r, D);
+end);
+
+# helper function for DigraphFromDreadnautString
+# parses partitions and stores them as vertex labels
+# returns nothing
+BindGlobal("DIGRAPHS_ParsePartition",
+function(r, minus, D)
+    local c, x, tempNewline, v1, v2, part, partition;
+
+    part := 1;
+    if r.n = fail then
+        ErrorNoReturn("Vertex number must be declared ",
+                    "before partition on line ", r.newline);
+    fi;
+    partition := List([1 .. r.n], x -> 0);
+
+    if minus then
+        return;
+    fi;
+
+    c := DIGRAPHS_GETNWCL(r, D, " ,\t");
+    if c = '=' then
+        c := DIGRAPHS_GETNWCL(r, D, " ,\t");
+    fi;
+
+    if IsDigitChar(c) then
+        DIGRAPHS_GetUngetChar(r, D, -1);
+        v1 := DIGRAPHS_readinteger(r, D);
+        if v1 - r.labelorg + 1 < 1 or v1 - r.labelorg + 1 > r.n then
+          Info(InfoWarning, 1, "Ignoring illegal vertex in partition", v1,
+              " on line ", r.newline);
+        else
+          partition[v1] := 1;
+        fi;
+        r.partition := partition;
+        return;
+    elif c <> '[' then
+        ErrorNoReturn("Partitions should be specified",
+                    " using one of the following formats:\n",
+                      "  * A single number (e.g. 'f=3')\n",
+                      "  * A list of cells (e.g. 'f=[...]')",
```

Comment:
Seems like the `\n` is missing here, but my comment is that I think error messages shouldn't have `\n` in them at all.